static_src/actions/user_actions.js

```diff
@@ -296,52 +296,55 @@ const userActions = {
   },
 
   createUserAndAssociate(userGuid) {
-    let entityGuid;
     const entityType = UserStore.currentlyViewedType;
+    const orgGuid = OrgStore.currentOrgGuid;
+    let cfApiRequest;
+    let entityGuid;
 
     if (entityType === ORG_NAME) {
-      entityGuid = OrgStore.currentOrgGuid;
+      entityGuid = orgGuid;
+      cfApiRequest = cfApi.putAssociateUserToOrganization.bind(cfApi, userGuid, entityGuid);
+
+      AppDispatcher.handleViewAction({
+        type: userActionTypes.USER_ORG_ASSOCIATE,
+        userGuid,
+        entityType,
+        entityGuid
+      });
     } else {
       entityGuid = SpaceStore.currentSpaceGuid;
+      cfApiRequest = cfApi.putAssociateUserToSpace.bind(cfApi, userGuid, orgGuid, entityGuid);
     }
 
-    AppDispatcher.handleViewAction({
-      type: userActionTypes.USER_ORG_ASSOCIATE,
-      userGuid,
-      entityType,
-      entityGuid
-    });
-
-    return cfApi.putAssociateUserToEntity(userGuid, entityGuid, entityType)
+    return cfApiRequest()
       .then(() => userActions.fetchEntityUsers(entityGuid, entityType))
-      .then(entityUsers => userActions.createdUserAndAssociated(userGuid, entityGuid, entityUsers));
+      .then(entityUsers => {
+        userActions.createdUserAndAssociated(userGuid, entityGuid, entityUsers, entityType);
+      });
   },
 
   fetchEntityUsers(entityGuid, entityType) {
-    let entityUsers;
-    if (entityType === ORG_NAME) {
-      entityUsers = cfApi.fetchOrgUsers(entityGuid);
-    } else {
-      entityUsers = cfApi.fetchSpaceUserRoles(entityGuid);
-    }
-    return Promise.resolve(entityUsers);
+    return cfApi[(entityType === ORG_NAME) ? 'fetchOrgUsers' : 'fetchSpaceUserRoles'](entityGuid);
   },
 
-  createdUserAndAssociated(userGuid, entityGuid, entityUsers) {
-    const user = entityUsers.filter((entityUser) => entityUser.guid === userGuid);
+  createdUserAndAssociated(userGuid, entityGuid, entityUsers, entityType) {
+    const user = entityUsers.filter(entityUser => entityUser.guid === userGuid);
 
-    if (!user[0]) {
+    if (!user.length) {
```

Comment:
Hmm I'm not sure I understand. If the array has a length of zero it is going to evaluate as false in the if statement.